ocaml/sdk-gen/go/gen_go_helper.ml

```diff
@@ -0,0 +1,274 @@
+  type enum = (string * string) list
+
+  module StringMap = Map.Make (String)
+
+  type enums = enum StringMap.t
+
+  let choose_enum _key a _b = Some a
+
+  let rec string_of_ty_with_enums ty : string * enums =
+    match ty with
+    | SecretString | String ->
+        ("string", StringMap.empty)
+    | Int ->
+        ("int", StringMap.empty)
+    | Float ->
+        ("float64", StringMap.empty)
+    | Bool ->
+        ("bool", StringMap.empty)
+    | DateTime ->
+        ("time.Time", StringMap.empty)
+    | Enum (name, kv) ->
+        let name = snake_to_camel name in
+        (name, StringMap.singleton name kv)
+    | Set ty ->
+        let s, e = string_of_ty_with_enums ty in
+        ("[]" ^ s, e)
+    | Map (ty1, ty2) ->
+        let s1, e1 = string_of_ty_with_enums ty1 in
+        let s2, e2 = string_of_ty_with_enums ty2 in
+        let ty = "map[" ^ s1 ^ "]" ^ s2 in
+        (ty, StringMap.union choose_enum e1 e2)
+    | Ref r ->
+        (snake_to_camel r ^ "Ref", StringMap.empty)
+    | Record r ->
+        (snake_to_camel r ^ "Record", StringMap.empty)
+    | Option ty ->
+        string_of_ty_with_enums ty
+
+  let of_enum name vs =
+    let name = snake_to_camel name in
+    let of_value (v, d) =
+      `O
+        [
+          ("value", `String v)
+        ; ("doc", `String d)
+        ; ("name", `String (name ^ snake_to_camel v))
+        ; ("type", `String name)
+        ]
+    in
+    `O [("name", `String name); ("values", `A (List.map of_value vs))]
+
+  let fields_of_obj_with_enums obj =
+    let rec flatten_contents contents =
+      List.fold_left
+        (fun l -> function
+          | Field f ->
+              f :: l
+          | Namespace (_name, contents) ->
+              flatten_contents contents @ l
+        )
+        [] contents
+    in
+    let fields = flatten_contents obj.contents in
+    let concat_and_convert field =
+      let concated =
+        String.concat "" (List.map snake_to_camel field.full_name)
+      in
+      match concated with
+      | "Uuid" | "Id" ->
+          String.uppercase_ascii concated
+      | _ ->
+          concated
+    in
+    List.fold_left
+      (fun (fields, enums, has_time_type) field ->
+        let is_time_type =
+          match field.ty with DateTime -> true | _ -> false
+        in
+        let ty, e = string_of_ty_with_enums field.ty in
+        ( `O
+            [
+              ("name", `String (concat_and_convert field))
+            ; ("description", `String (String.trim field.field_description))
+            ; ("type", `String ty)
+            ]
+          :: fields
+        , StringMap.union choose_enum enums e
+        , if is_time_type then true else has_time_type
+        )
+      )
+      ([], StringMap.empty, false)
+      fields
+
+  let enums_from_result obj msg =
+    match msg.msg_result with
+    | None ->
+        StringMap.empty
+    | Some (t, _d) ->
+        if obj.name = "event" && String.lowercase_ascii msg.msg_name = "from"
```

Comment:
Why we need to care about this special case?
Would `string_of_ty_with_enums` return just `StringMap.empty`?